src/components/schedule-inputs.tsx

```diff
@@ -252,3 +489,97 @@ export function ScheduleInputs(props: ScheduleInputsProps): JSX.Element | null {
     </>
   );
 }
+
+function parseTime(input: string): [number | undefined, number | undefined] {
+  // Allow h:mm or hh:mm
+  const timeRegex = /^(\d\d?):(\d\d)$/;
+  const timeResult = timeRegex.exec(input);
+
+  let hours: number | undefined = undefined;
+  let minutes: number | undefined = undefined;
+
+  if (timeResult) {
+    hours = parseInt(timeResult[1]);
+    minutes = parseInt(timeResult[2]);
+  }
+
+  return [hours, minutes];
+}
+
+/**
+ * Accepts a translation bundle in its constructor and returns the appropriate
+ * error message. If no errors are present, the methods return an empty string.
+ */
+export class ScheduleValidator {
```

Comment:
This file is getting a little sprawling; should `ScheduleValidator` be moved to a source file of its own?

Reply:
If it's not being used anywhere else, I think it's best to keep it local to the file it's being used in. IMO, traversing long import paths is worse than having big files in terms of readability and maintainability. Open to feedback on this.